player.py

```diff
@@ -32,19 +32,26 @@ def make_move(self, state):
 
         pawn = None
         valid = False
-        print('Begin by selecting a pawn in a row and column')
-        while not valid:
-            row, col = eval(input('row, col: '))
-            pawn = state.get_pawn(row, col, self.color)
-            if pawn is not None:
-                valid = True
         final_pos = None
-        valid = False
-        print('Select a final position for the pawn')
         while not valid:
-            row, col = eval(input('row, col: '))
-            if state.check_cell(row, col) == label.BLANK:
-                final_pos = Position(row, col)
-                valid = True
+            print('Begin by selecting a pawn in a row and column')
+            row = eval(input('Select a row: '))
+            col = eval(input('Select a column: '))
+            if (row in range(3)) and (col in range(3)):
+                pawn = state.get_pawn(row, col, self.color)
+                if pawn is not None:
+                    valid = True
+            print('Select a final position for the pawn')
+            row = eval(input('Select a row: '))
+            col = eval(input('Select a column: '))
+            for moveset in state.get_available_moves():
+                if repr(Move(pawn, Position(row, col))) == repr(moveset):
```

Comment:
Bug? I believe that this proposition is always false because: (1) the left-hand side represents a move `Move(...)` and (2) the right-hand side represents a _list_ of moves `[...]`. So prefix and suffix should never match.

I believe the intent of this for-loop is to: (1) validate the user move and (2) construct the position. Perhaps we can make this more concise as follows.
```python
final_pos = Position(row, col)
valid = Move(pawn, final_pos)) in state.get_available_moves()
```

Thoughts?

Reply:
This may stem from my confusing labeling of the 'for'-loop:

get_available_moves ; "gives a **list** of all the pairs of original and final positions that are legally allowed for the player".
I thought of these pairs of original and final positions as sets of moves so I named each element in the 'for'-loop, 'moveset'.

Hence, 
moveset; "A single pair of original and final positions that constructs a move".

So if the player selects for example: B at (2, 2) (1, 2)
the loop will confirm that it is a match for one of the elements (moveset) of the get_available_moves list

I can change the elements to something more clear like 'movepairs' for example

Reply:
You are correct. :)

I had misinterpreted the `moveset` as a list without running the code. I like the idea that `movepairs` would avoid ambiguity.

I also think that we could also cut down on several levels of indentation and simplify the logic by using the previous proposition (or some variant):

```python
final_pos = Position(row, col)
valid = Move(pawn, final_pos)) in state.get_available_moves()
```

Can you think of any other ways to simplify this code?

Reply:
Yes, this works wonderfully!
I originally tried to use something along these lines but couldn't get it to work until I used repr() but at that point I had already overcomplicated the process with the for-loop. Yes, this is much better.